### PR TITLE
WIP: Erste Version Reporting

### PR DESCRIPTION
**Pull request by nils-tekampe**

### CODE REVIEW

File: src/main/java/de/konfidas/ttc/reporting/Report.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Generic Class for reporting. The class represents a tree structure of elements
+ * that make up a report
+ * @param <T> The type of the data to be reported
+ */
+public class Report<T> {
```
Comment (dheldt) [inline src/main/java/de/konfidas/ttc/reporting/Report.java:11]: You should never implement basic datatypes yourself (like lists, trees, etc.). 

this is a simple tree with children and a generic datatype. there has to be some class for this in the java api to use. there is no need to do this on your own.

Comment (nils-tekampe) [reply src/main/java/de/konfidas/ttc/reporting/Report.java:11]: I did not find any implementation in the Java framework for a tree (surprised me as well). Do you have a specific class in mind?

File: pom.xml [hunk 2]
```diff
@@ -9,6 +9,11 @@
     <artifactId>TTC</artifactId>
     <version>0.2-PRE-ALPHA</version>
     <dependencies>
+        <dependency>
```
Comment (dheldt) [inline pom.xml:12]: was this intentionally? if so, why?

File: src/main/java/de/konfidas/ttc/reporting/ReportTextPrinter.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package de.konfidas.ttc.reporting;
+
+import de.konfidas.ttc.messages.LogMessage;
+import de.konfidas.ttc.messages.TransactionLogMessage;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.StringUtils;
+import org.bouncycastle.asn1.ASN1Primitive;
+
+public class ReportTextPrinter {
```
Comment (dheldt) [inline src/main/java/de/konfidas/ttc/reporting/ReportTextPrinter.java:9]: I think, we should not split up the report, the reporter and the printer of the report? so we have as input a

Collection<LogMessages> and Collection<ValidationException>, next, it is the reporters responsibility to do something with this, for example, create one XML, where all comes together... or some junit-like output. But I do not see the need for an intermediate "report", which can than again be printed into different formats. I guesss you had some reason to do so, could you explain what I am missing?

File: src/main/java/de/konfidas/ttc/reporting/Report.java (hunk 1 above)
Comment (dheldt) [reply src/main/java/de/konfidas/ttc/reporting/Report.java:11]: I have to admit i do not. and a quick search did not bring up something useful.

**Pull request merged**